hcpannot/analysis/contour.py

```diff
@@ -0,0 +1,253 @@
+"""
+    Codes for averaging contours of the same subject among researchers
+    and plotting the contours on flatmaps
+    
+"""
+
+from hcpannot.proc import (proc, rigid_align_points)
+from hcpannot.config import (ventral_raters, meanrater)
+import neuropythy as ny
+import os
+import matplotlib.pyplot as plt
+from matplotlib.collections import LineCollection
+import numpy as np
+
+# list of raters
+raters = ventral_raters
+raters.append(meanrater)
+
+# generate a list of colors
+colors = ['r', 'g', 'b', 'c', 'm', 'k']
+rater_colors = {r: c for r, c in zip(raters, colors)}
+
+
+def nestget(d, k):
+    """Retrieves nested data from the proc dictionaries.
+    
+    Certain keys such as `'boundaries'` are accessible in the dictionaries that
+    are returned by the `proc` function only via the `'nested_data'` key, which
+    typically contains another proc dictionary with additional data. The
+    `nestget` function gets data from these embedded dictionaries.
+    """
+    while k not in d:
+        d = d['nested_data']
+    return d[k]
+
+def gen_contour_coords(rater, subject_id, hemi, contour, proc_path, contour_save_path, npoints=500):
+    """
+    Divide a contour of a given subject, rater, and hemisphere into a given number
+    (default 500) of points and save the coordinates.
+    """
+    
+    # check if the contour already exists, if so, skip the processing
+    cache_file = os.path.join(proc_path, 'fsaverage', f'cacherater_{rater}_{subject_id}_{hemi}_{contour}.mgz')
+    
+    data = proc('ventral', rater=rater, sid=subject_id, hemisphere=hemi, 
+               save_path=proc_path, load_path=contour_save_path)
+    
+    if contour != 'V3v': 
+        trace = data['traces'][contour]
+        # break the contour into evenly spaced points (default is 500)
+        coords = trace.curve.linspace(npoints)
+    
+    # V3v was not annotated by researchers in this project, and the data are stored differently
+    else:
+        v3v = nestget(data, 'v3v_contour')
+        v3v_curve = ny.curve_spline(v3v[0], v3v[1])
+        coords = v3v_curve.linspace(npoints)
+        
+    # align the points to fsaverage space
+    cortex = data['cortex']
+    fmap1 = data['flatmap']
+    fmap2 = ny.to_flatmap('occipital_pole', cortex)
+    addr = fmap1.address(coords)
+    coordinates = fmap2.unaddress(addr)
+    
+    return coordinates
+
+def plot_contours(raters, subject_id, hemi, contours, 
```

Comment:
Just a few things:

1. Review to make sure that we use proc data instead of the cacherater files.
2. Maybe rename since there is another `plot_contours` function in the repo—make this more explicit like `plot_subject_contours_across_raters` (probably too long, something like that).
3. I would change `proc_path` to `save_path` and `contour_save_path` to `data_path` to be consistent withte